js/ojsxc.js

```diff
@@ -392,4 +398,63 @@
             observeContactsMenu();
         }
     });
+
+    $(document).on( "click", '#jsxc_roster p', function() {
+        if (jsxc.storage.getItem('serverType') === 'internal') {
+           startInternalBackend();
+        }
+    });
+
+    function startInternalBackend() {
+       jsxc.bid = OC.currentUser.toLowerCase() + '@' + window.location.host;
+
+        jsxc.storage.setItem('jid', jsxc.bid + '/internal');
+        jsxc.storage.setItem('sid', 'internal');
+        jsxc.storage.setItem('rid', '123456');
+
+        jsxc.options.set('xmpp', {
+            url: OC.generateUrl('apps/ojsxc/http-bind')
+        });
+
+        jsxc.start(jsxc.bid + '/internal', 'internal', '123456');
+        jsxc.gui.restore();
+        jsxc.gui.roster.toggle(jsxc.CONST.SHOWN);
+        $(document).trigger('attached.jsxc');
```

Comment:
The above 3 lines should be handled by `jsxc.start`, but as above I will check it.